app/models/plugins.rb

```diff
@@ -4,9 +4,12 @@ def table_name_prefix
       'plugins_'
     end
 
-    def create_for_page(plugin, page)
-      plugin = plugin.new(plugin.const_get(:DEFAULTS))
+    def create_for_page(plugin_name, page, ref)
+      return true if plugin_name.blank? || page.blank?
+      plugin_class = Plugins.const_get(plugin_name.camelcase)
+      plugin = plugin_class.new(plugin_class.const_get(:DEFAULTS))
```

Comment:
This is where the weird bug is. Sometimes `Plugins.const_get('Action')` returns `Plugins::Action` and sometimes it returns `Action`.